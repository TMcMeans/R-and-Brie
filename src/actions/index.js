export const addRecipes = recipes => ({
  type: 'ADD_RECIPES',
  recipes: recipes
});

export const hasErrored = message => ({
  type: 'HAS_ERRORED',
  message
});

export const isLoading = bool => ({
  type: 'IS_LOADING',
  isLoading: bool
});

export const addReview = review => ({
  type: 'ADD_REVIEW',
  review: review
})