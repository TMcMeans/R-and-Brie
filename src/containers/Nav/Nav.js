import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Nav extends Component {
  constructor() {
    super();
    this.state = {
      selectedCheese: ''
    };
  }

  render() {
    return (
      <div className="nav">
        <Link className="nav-item" to="/recipes">
          <div className="cheese">
            <img src="" alt="" id="brie" />
          </div>
          Brie
        </Link>
        <Link className="nav-item" to="/recipes">
          <div className="cheese">
            <img src="" alt="" id="parmesean" />
          </div>
          Parmesan
        </Link>
        <Link className="nav-item" to="/recipes">
          <div className="cheese">
            <img src="" alt="" id="feta" />
          </div>
          Feta
        </Link>
        <Link className="nav-item" to="/recipes">
          <div className="cheese">
            <img src="" alt="" id="swiss" />
          </div>
          Swiss
        </Link>
        <Link className="nav-item" to="/recipes">
          <div className="cheese">
            <img src="" alt="" id="mozzarella" />
          </div>
          Mozzarella
        </Link>
        <Link className="nav-item" to="/recipes">
          <div className="cheese">
            <img src="" alt="" id="cheddar" />
          </div>
          Cheddar
        </Link>
        <Link className="nav-item" to="/recipes">
          <div className="cheese">
            <img src="" alt="" id="blue" />
          </div>
          Blue
        </Link>
        <Link className="nav-item" to="/recipes">
          <div className="cheese">
            <img src="" alt="" id="asiago" />
          </div>
          Asiago
        </Link>
        <Link className="nav-item" to="/recipes">
          <div className="cheese">
            <img src="" alt="" id="fontina" />
          </div>
          Fontina
        </Link>
        <Link className="nav-item" to="/recipes">
          <div className="cheese">
            <img src="" alt="" id="american" />
          </div>
          American
        </Link>
      </div>
    );
  }
}

export default Nav;
