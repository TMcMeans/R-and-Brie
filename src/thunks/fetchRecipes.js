import { addRecipes, hasErrored, isLoading } from '../actions';

export const fetchRecipes = url => {
  return async dispatch => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      dispatch(isLoading(false));
      const data = await response.json();
      const recipes = data.hits;
      dispatch(addRecipes(recipes));
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  };
};
