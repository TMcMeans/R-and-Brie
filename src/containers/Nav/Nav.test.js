import React from 'react'
import { shallow } from 'enzyme'
import { Nav, mapStateToProps, mapDispatchToProps } from './Nav'
import { addRecipes } from '../../actions'
import { fetchRecipes } from '../../thunks/fetchRecipes'

describe('Nav', () => {
  it('should render snapshot with correct data passed in', () => {
    const mockFunc = jest.fn()
    const wrapper = shallow(<Nav recipes={[]} isLoading={false}
      fetchRecipes={mockFunc} addRecipes={mockFunc} />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return an array of recipes', () => {
      const mockState = {
        recipes: [{ label: 'baked brie' }, { label: 'brie burgers' }],
        isLoading: false
      }

      const expected = {
        recipes: [{ label: 'baked brie' }, { label: 'brie burgers' }],
        isLoading: false
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)

    })
    it('should return a boolean for isLoading', () => {
      const mockState = {
        recipes: [],
        isLoading: true
      }

      const expected = {
        recipes: [],
        isLoading: true
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
  describe('mapDispatchToProps', () => {

    const mockDispatch = jest.fn()

    it('should dispatch fetchRecipes when updateRecipes is called', () => {
      const mockUrl = 'someplace.com'
      const actionToDispatch = fetchRecipes(mockUrl)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchRecipes(mockUrl)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)

    })

    it('should dispatch addRecipes when updateRecipes is called', () => {
      const actionToDispatch = addRecipes([])

      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.addRecipes([])
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})

