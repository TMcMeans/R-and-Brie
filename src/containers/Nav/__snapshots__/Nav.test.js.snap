// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render snapshot with correct data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    addRecipes={[MockFunction]}
    fetchRecipes={[MockFunction]}
    isLoading={false}
    recipes={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="nav-item"
          onClick={[Function]}
          replace={false}
          to="/type/brie"
        >
          <div
            className="cheese"
          >
            <img
              alt=""
              id="brie"
              src=""
            />
          </div>
          Brie
        </Link>,
        <Link
          className="nav-item"
          onClick={[Function]}
          replace={false}
          to="/type/parmesean"
        >
          <div
            className="cheese"
          >
            <img
              alt=""
              id="parmesean"
              src=""
            />
          </div>
          Parmesan
        </Link>,
        <Link
          className="nav-item"
          onClick={[Function]}
          replace={false}
          to="/type/feta"
        >
          <div
            className="cheese"
          >
            <img
              alt=""
              id="feta"
              src=""
            />
          </div>
          Feta
        </Link>,
        <Link
          className="nav-item"
          onClick={[Function]}
          replace={false}
          to="/type/swiss"
        >
          <div
            className="cheese"
          >
            <img
              alt=""
              id="swiss"
              src=""
            />
          </div>
          Swiss
        </Link>,
        <Link
          className="nav-item"
          onClick={[Function]}
          replace={false}
          to="/type/mozzarella"
        >
          <div
            className="cheese"
          >
            <img
              alt=""
              id="mozzarella"
              src=""
            />
          </div>
          Mozzarella
        </Link>,
        <Link
          className="nav-item"
          onClick={[Function]}
          replace={false}
          to="/type/cheddar"
        >
          <div
            className="cheese"
          >
            <img
              alt=""
              id="cheddar"
              src=""
            />
          </div>
          Cheddar
        </Link>,
        <Link
          className="nav-item"
          onClick={[Function]}
          replace={false}
          to="/type/blue%20cheese"
        >
          <div
            className="cheese"
          >
            <img
              alt=""
              id="bluecheese"
              src=""
            />
          </div>
          Blue
        </Link>,
        <Link
          className="nav-item"
          onClick={[Function]}
          replace={false}
          to="/type/asiago"
        >
          <div
            className="cheese"
          >
            <img
              alt=""
              id="asiago"
              src=""
            />
          </div>
          Asiago
        </Link>,
        <Link
          className="nav-item"
          onClick={[Function]}
          replace={false}
          to="/type/fontina"
        >
          <div
            className="cheese"
          >
            <img
              alt=""
              id="fontina"
              src=""
            />
          </div>
          Fontina
        </Link>,
        <Link
          className="nav-item"
          onClick={[Function]}
          replace={false}
          to="/type/american%20cheese"
        >
          <div
            className="cheese"
          >
            <img
              alt=""
              id="american"
              src=""
            />
          </div>
          American
        </Link>,
      ],
      "className": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="cheese"
            >
              <img
                alt=""
                id="brie"
                src=""
              />
            </div>,
            "Brie",
          ],
          "className": "nav-item",
          "onClick": [Function],
          "replace": false,
          "to": "/type/brie",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                id="brie"
                src=""
              />,
              "className": "cheese",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "id": "brie",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "Brie",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="cheese"
            >
              <img
                alt=""
                id="parmesean"
                src=""
              />
            </div>,
            "Parmesan",
          ],
          "className": "nav-item",
          "onClick": [Function],
          "replace": false,
          "to": "/type/parmesean",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                id="parmesean"
                src=""
              />,
              "className": "cheese",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "id": "parmesean",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "Parmesan",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="cheese"
            >
              <img
                alt=""
                id="feta"
                src=""
              />
            </div>,
            "Feta",
          ],
          "className": "nav-item",
          "onClick": [Function],
          "replace": false,
          "to": "/type/feta",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                id="feta"
                src=""
              />,
              "className": "cheese",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "id": "feta",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "Feta",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="cheese"
            >
              <img
                alt=""
                id="swiss"
                src=""
              />
            </div>,
            "Swiss",
          ],
          "className": "nav-item",
          "onClick": [Function],
          "replace": false,
          "to": "/type/swiss",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                id="swiss"
                src=""
              />,
              "className": "cheese",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "id": "swiss",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "Swiss",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="cheese"
            >
              <img
                alt=""
                id="mozzarella"
                src=""
              />
            </div>,
            "Mozzarella",
          ],
          "className": "nav-item",
          "onClick": [Function],
          "replace": false,
          "to": "/type/mozzarella",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                id="mozzarella"
                src=""
              />,
              "className": "cheese",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "id": "mozzarella",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "Mozzarella",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="cheese"
            >
              <img
                alt=""
                id="cheddar"
                src=""
              />
            </div>,
            "Cheddar",
          ],
          "className": "nav-item",
          "onClick": [Function],
          "replace": false,
          "to": "/type/cheddar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                id="cheddar"
                src=""
              />,
              "className": "cheese",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "id": "cheddar",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "Cheddar",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="cheese"
            >
              <img
                alt=""
                id="bluecheese"
                src=""
              />
            </div>,
            "Blue",
          ],
          "className": "nav-item",
          "onClick": [Function],
          "replace": false,
          "to": "/type/blue%20cheese",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                id="bluecheese"
                src=""
              />,
              "className": "cheese",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "id": "bluecheese",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "Blue",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="cheese"
            >
              <img
                alt=""
                id="asiago"
                src=""
              />
            </div>,
            "Asiago",
          ],
          "className": "nav-item",
          "onClick": [Function],
          "replace": false,
          "to": "/type/asiago",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                id="asiago"
                src=""
              />,
              "className": "cheese",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "id": "asiago",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "Asiago",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="cheese"
            >
              <img
                alt=""
                id="fontina"
                src=""
              />
            </div>,
            "Fontina",
          ],
          "className": "nav-item",
          "onClick": [Function],
          "replace": false,
          "to": "/type/fontina",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                id="fontina"
                src=""
              />,
              "className": "cheese",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "id": "fontina",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "Fontina",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="cheese"
            >
              <img
                alt=""
                id="american"
                src=""
              />
            </div>,
            "American",
          ],
          "className": "nav-item",
          "onClick": [Function],
          "replace": false,
          "to": "/type/american%20cheese",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                id="american"
                src=""
              />,
              "className": "cheese",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "id": "american",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "American",
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="nav-item"
            onClick={[Function]}
            replace={false}
            to="/type/brie"
          >
            <div
              className="cheese"
            >
              <img
                alt=""
                id="brie"
                src=""
              />
            </div>
            Brie
          </Link>,
          <Link
            className="nav-item"
            onClick={[Function]}
            replace={false}
            to="/type/parmesean"
          >
            <div
              className="cheese"
            >
              <img
                alt=""
                id="parmesean"
                src=""
              />
            </div>
            Parmesan
          </Link>,
          <Link
            className="nav-item"
            onClick={[Function]}
            replace={false}
            to="/type/feta"
          >
            <div
              className="cheese"
            >
              <img
                alt=""
                id="feta"
                src=""
              />
            </div>
            Feta
          </Link>,
          <Link
            className="nav-item"
            onClick={[Function]}
            replace={false}
            to="/type/swiss"
          >
            <div
              className="cheese"
            >
              <img
                alt=""
                id="swiss"
                src=""
              />
            </div>
            Swiss
          </Link>,
          <Link
            className="nav-item"
            onClick={[Function]}
            replace={false}
            to="/type/mozzarella"
          >
            <div
              className="cheese"
            >
              <img
                alt=""
                id="mozzarella"
                src=""
              />
            </div>
            Mozzarella
          </Link>,
          <Link
            className="nav-item"
            onClick={[Function]}
            replace={false}
            to="/type/cheddar"
          >
            <div
              className="cheese"
            >
              <img
                alt=""
                id="cheddar"
                src=""
              />
            </div>
            Cheddar
          </Link>,
          <Link
            className="nav-item"
            onClick={[Function]}
            replace={false}
            to="/type/blue%20cheese"
          >
            <div
              className="cheese"
            >
              <img
                alt=""
                id="bluecheese"
                src=""
              />
            </div>
            Blue
          </Link>,
          <Link
            className="nav-item"
            onClick={[Function]}
            replace={false}
            to="/type/asiago"
          >
            <div
              className="cheese"
            >
              <img
                alt=""
                id="asiago"
                src=""
              />
            </div>
            Asiago
          </Link>,
          <Link
            className="nav-item"
            onClick={[Function]}
            replace={false}
            to="/type/fontina"
          >
            <div
              className="cheese"
            >
              <img
                alt=""
                id="fontina"
                src=""
              />
            </div>
            Fontina
          </Link>,
          <Link
            className="nav-item"
            onClick={[Function]}
            replace={false}
            to="/type/american%20cheese"
          >
            <div
              className="cheese"
            >
              <img
                alt=""
                id="american"
                src=""
              />
            </div>
            American
          </Link>,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="cheese"
              >
                <img
                  alt=""
                  id="brie"
                  src=""
                />
              </div>,
              "Brie",
            ],
            "className": "nav-item",
            "onClick": [Function],
            "replace": false,
            "to": "/type/brie",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  id="brie"
                  src=""
                />,
                "className": "cheese",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "id": "brie",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "Brie",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="cheese"
              >
                <img
                  alt=""
                  id="parmesean"
                  src=""
                />
              </div>,
              "Parmesan",
            ],
            "className": "nav-item",
            "onClick": [Function],
            "replace": false,
            "to": "/type/parmesean",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  id="parmesean"
                  src=""
                />,
                "className": "cheese",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "id": "parmesean",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "Parmesan",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="cheese"
              >
                <img
                  alt=""
                  id="feta"
                  src=""
                />
              </div>,
              "Feta",
            ],
            "className": "nav-item",
            "onClick": [Function],
            "replace": false,
            "to": "/type/feta",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  id="feta"
                  src=""
                />,
                "className": "cheese",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "id": "feta",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "Feta",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="cheese"
              >
                <img
                  alt=""
                  id="swiss"
                  src=""
                />
              </div>,
              "Swiss",
            ],
            "className": "nav-item",
            "onClick": [Function],
            "replace": false,
            "to": "/type/swiss",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  id="swiss"
                  src=""
                />,
                "className": "cheese",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "id": "swiss",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "Swiss",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="cheese"
              >
                <img
                  alt=""
                  id="mozzarella"
                  src=""
                />
              </div>,
              "Mozzarella",
            ],
            "className": "nav-item",
            "onClick": [Function],
            "replace": false,
            "to": "/type/mozzarella",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  id="mozzarella"
                  src=""
                />,
                "className": "cheese",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "id": "mozzarella",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "Mozzarella",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="cheese"
              >
                <img
                  alt=""
                  id="cheddar"
                  src=""
                />
              </div>,
              "Cheddar",
            ],
            "className": "nav-item",
            "onClick": [Function],
            "replace": false,
            "to": "/type/cheddar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  id="cheddar"
                  src=""
                />,
                "className": "cheese",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "id": "cheddar",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "Cheddar",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="cheese"
              >
                <img
                  alt=""
                  id="bluecheese"
                  src=""
                />
              </div>,
              "Blue",
            ],
            "className": "nav-item",
            "onClick": [Function],
            "replace": false,
            "to": "/type/blue%20cheese",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  id="bluecheese"
                  src=""
                />,
                "className": "cheese",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "id": "bluecheese",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "Blue",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="cheese"
              >
                <img
                  alt=""
                  id="asiago"
                  src=""
                />
              </div>,
              "Asiago",
            ],
            "className": "nav-item",
            "onClick": [Function],
            "replace": false,
            "to": "/type/asiago",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  id="asiago"
                  src=""
                />,
                "className": "cheese",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "id": "asiago",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "Asiago",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="cheese"
              >
                <img
                  alt=""
                  id="fontina"
                  src=""
                />
              </div>,
              "Fontina",
            ],
            "className": "nav-item",
            "onClick": [Function],
            "replace": false,
            "to": "/type/fontina",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  id="fontina"
                  src=""
                />,
                "className": "cheese",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "id": "fontina",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "Fontina",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="cheese"
              >
                <img
                  alt=""
                  id="american"
                  src=""
                />
              </div>,
              "American",
            ],
            "className": "nav-item",
            "onClick": [Function],
            "replace": false,
            "to": "/type/american%20cheese",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  id="american"
                  src=""
                />,
                "className": "cheese",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "id": "american",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "American",
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
